ToyBox/classes/MainUI/Browser/FactsEditor.cs

```diff
@@ -90,7 +91,9 @@ List<Action> todo
 
             var lastRect = GUILayoutUtility.GetLastRect();
 
-            var mutatorLookup = BlueprintAction.ActionsForType(typeof(Definition)).Distinct().ToDictionary(a => a.name, a => a);
+            var mutatorLookup = BlueprintAction.ActionsForType(blueprint.GetType())
+                .GroupBy(a => a.name).Select(g => g.FirstOrDefault())
+                .ToDictionary(a => a.name, a => a);
```

Comment:
`Distinct` doesn't work in this case since that compares the objects directly and we care about the object's `name` property.
`.GroupBy(a => a.name).Select(g => g.FirstOrDefault())` then filters out duplicates based on name.

Reply:
The original code creates a dictionary with `BlueprintAction.name` as the key.

Problem is `Distinct()` fails to ensure names are unique and thus will throw an exception if we get multiple BlueprintActions with the same name, while `.GroupBy(a => a.name).Select(g => g.FirstOrDefault())` will ensure that `BlueprintAction` objects with duplicate names will be pruned.